packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -2230,7 +2230,7 @@ class _SemanticsSortGroup extends Comparable<_SemanticsSortGroup> {
     final List<_BoxEdge> edges = <_BoxEdge>[];
     for (SemanticsNode child in nodes) {
       // Using a small delta to shrink child rects removes overlapping cases.
-      final Rect childRect = child.rect.deflate(0.1);
+      final Rect childRect = child.rect.deflate(1e-15);
```

Comment:
My thought with this is that removing a trivially small amount like this should still make them not evaluate as overlapping.

Reply:
Our original thought here was that even when things are overlapping visibly, if the overlap is smallish, it shouldn't count as overlapping...

Reply:
That ends up breaking traversal order :(

Reply:
So what's the vote here?  Add the very small deflate, or remove it entirely and add a TODO?

Reply:
The canonical example is the categories on the flutter gallery home page. It should go left to right, but before it was going top to bottom.

If it still goes left to right with this change I'm okay with it. If it doesn't, we might as well remove the deflate and go back to the drawing board

Reply:
The categories sample still worked with this change, but it's actually the traversal order we want to change (which was going top to bottom instead of left to right).  I've updated that test to go left to right and put back the deflate to 0.1.